modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/OrderResourceImpl.java

```diff
@@ -546,16 +559,26 @@ private CommerceOrder _updateNestedResources(
 				_getOrderItemExternalReferenceCodes(orderItems));
 
 			for (OrderItem orderItem : orderItems) {
-				OrderItemUtil.addOrUpdateCommerceOrderItem(
-					_cpInstanceService, _commerceOrderItemService,
-					_commerceOrderModelResourcePermission, orderItem,
-					commerceOrder,
-					_commerceContextFactory.create(
-						contextCompany.getCompanyId(),
-						commerceOrder.getGroupId(), contextUser.getUserId(),
-						commerceOrder.getCommerceOrderId(),
-						commerceOrder.getCommerceAccountId()),
-					serviceContext);
+				CommerceOrderItem commerceOrderItem =
+					OrderItemUtil.addOrUpdateCommerceOrderItem(
+						_cpInstanceService, _commerceOrderItemService,
+						_commerceOrderModelResourcePermission, orderItem,
+						commerceOrder,
+						_commerceContextFactory.create(
+							contextCompany.getCompanyId(),
+							commerceOrder.getGroupId(), contextUser.getUserId(),
+							commerceOrder.getCommerceOrderId(),
+							commerceOrder.getCommerceAccountId()),
+						serviceContext);
```

Comment:
I'm not sure this will work if you don't instantiate a new ServiceContext, I remember we had a lot of bugs on CustomFields with headless APIs and the fix has always been reinstantiate the ServiceContext

Reply:
@andrea-ale-sbarra , requested change has been made

Reply:
thanks! let's wait for @luca-pellizzon review